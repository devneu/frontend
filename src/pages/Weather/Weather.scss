.weather-component {
  background: #DCDCDC;
  width: 100%;
  height: calc(100vh - 64px);

  .loader {
    position: absolute;
    top: 50%;
    transform: translate(0, -50%);
    width: 100%;
    padding: 10px;
  }

  .search-bar {
    margin: 0 auto;
    padding: 10px;
    width: 25%;
  }

  .weather-card {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: 18px;
    height: 240px;
    max-width: 178px;
    padding: 15px;
    margin: 0 auto;
    border: 3px solid;
    border-radius: 20%;

    .weather-temperature {
      font-size: 31px;
    }

  }

  .forecast {
    display: flex;
    flex-direction: row;
    height: 200px;
    width: 1135px;
    margin: 0 auto;

    .forecast-item {
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
      width: 122px;
      margin: 10px;
      padding: 10px;
      border: 2px solid whitesmoke;
      border-radius: 10%;
      font-size: 16px;
    }

  }

}
