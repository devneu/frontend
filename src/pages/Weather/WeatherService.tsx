import axios from 'axios';

export class WeatherService {
  // eslint-disable-next-line no-useless-constructor,no-empty-function
  constructor(private API_KEY: string = '0bf1d5f6e1ff530158a2e691a88eadc7') {}

  getWeather(city: string): Promise<any> {
    return axios
      .get(
        `https://api.openweathermap.org/data/2.5/weather?units=metric&q=${city}&appid=${this.API_KEY}`,
      )
      .then((response) => response.data)
      .catch((error) => {
        // eslint-disable-next-line no-console
        console.log(error);
      });
  }

  getWeatherForecast(): Promise<any> {
    return axios
      .get(
        `https://api.openweathermap.org/data/2.5/onecall?lat=49&lon=24&exclude=hourly,minutely&units=metric&appid=${this.API_KEY}`,
      )
      .then((response) => response.data)
      .catch((error) => {
        // eslint-disable-next-line no-console
        console.log(error);
      });
  }
}
const weatherServiceInstance = new WeatherService();
export default weatherServiceInstance;
