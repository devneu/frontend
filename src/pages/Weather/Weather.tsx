import React, { useState } from 'react';
import './Weather.scss';
import axios from 'axios';
import Loader from 'react-loader-spinner';
import 'react-loader-spinner/dist/loader/css/react-spinner-loader.css';
import { Input } from 'antd';

const Weather = () => {
  const [city, setCity] = useState('');
  const [loading, setLoaded] = useState(true);
  const [wdata, setWeather] = useState<any>(null);
  async function Api() {
    axios
      .get(
        `http://api.openweathermap.org/data/2.5/weather?units=metric&q=${city}&appid=fc6916988b0954c2540d0f09fb7e8a65`,
      )
      .then((response) => {
        // handle success
        console.log(response.data);
        setWeather(response.data);
        setLoaded(false);
        setCity('');
      })
      .catch((error) => {
        // handle error
        console.log(error);
      });
  }
  const search = (evt: any) => {
    if (evt.key === 'Enter') {
      Api();
    }
  };

  return (
    <div className="weather-components">
      <Loader visible={loading} type="Circles" color="#00BFFF" height={200} width={200} />
      <Input
        type="text"
        placeholder="Search"
        onChange={(e) => setCity(e.target.value)}
        value={city}
        onKeyPress={search}
      />
      {typeof wdata?.main !== 'undefined' ? (
        <div className="weather-block">
          <div className="weather-city">{wdata?.name}</div>
          <div className="weather-description">{wdata?.weather?.[0]?.description}</div>
          <img
            className="weather-icon"
            src={`https://openweathermap.org/img/wn/${wdata?.weather?.[0]?.icon}@2x.png`}
            alt="icon"
          />
          <div className="weather-temp">
            {`${wdata?.main?.temp}`}
            Â°c
          </div>
        </div>
      ) : (
        ''
      )}
    </div>
  );
};
export default Weather;
