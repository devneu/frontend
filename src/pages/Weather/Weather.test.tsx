import React from 'react';
import render from 'react-test-renderer';
import Weather from './Weather';

describe('Weather', () => {
  it('Should successfully render', () => {
    const wrapper = render.create(<Weather />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
