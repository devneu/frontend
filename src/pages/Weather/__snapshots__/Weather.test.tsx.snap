// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Should successfully render 1`] = `
<div
  className="weather-component"
>
  <div
    className="ant-spin ant-spin-lg ant-spin-spinning loader"
  >
    <span
      className="ant-spin-dot ant-spin-dot-spin"
    >
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
    </span>
  </div>
</div>
`;
